iroh/src/node/builder.rs

```diff
@@ -422,11 +426,22 @@ where
         self
     }
 
-    /// Binds the node service to a different socket.
-    ///
-    /// By default it binds to `127.0.0.1:11204`.
-    pub fn bind_port(mut self, port: u16) -> Self {
-        self.bind_port.replace(port);
+    /// Binds the node service to a specific socket IPv4 address.
+    pub fn bind_addr_v4(mut self, addr: SocketAddrV4) -> Self {
+        self.addr_v4 = addr;
+        self
+    }
+
+    /// Binds the node service to a specific socket IPv6 address.
+    pub fn bind_addr_v6(mut self, addr: SocketAddrV6) -> Self {
+        self.addr_v6 = addr;
+        self
+    }
+
+    /// Use a random port for both IPv4 and IPv6.
```

Comment:
```suggestion
    /// Use a random port for both IPv4 and IPv6.
    ///
    /// This is a convenience function useful when you do not need a specific port 
    /// and want to avoid conflicts when running multiple instances, e.g. in tests.
    ///
    /// This overrides the ports of the socket addresses provided by [`Builder::bind_addr_v4`]
    /// and [`Builder::bind_addr_v6`].  By default both of those bind to the
    /// unspecified address, which would result in `0.0.0.0:0` and `[::]:0` as bind
    /// addresses unless they are changed.
```

But to be honest, given the behaviour of the normal port binding.  Is there much value to having this?  Because in tests we will already just fall back to port `0` if another test is already using it.

Reply:
the iroh port is used by default, not `0` for `iroh` itself

Reply:
updated docs